pkg/admissioncontroller/webhook/auth/seed/graph/eventhandler_shoot.go

```diff
@@ -72,6 +73,34 @@ func (g *graph) setupShootWatch(ctx context.Context, informer cache.Informer) er
 	return err
 }
 
+func (g *graph) hasExpectedShootBindingEdges(shoot *gardencorev1beta1.Shoot) bool {
+	g.lock.RLock()
+	defer g.lock.RUnlock()
+
+	isEdgeMissingFrom := func(vertexType VertexType, bindingName *string) bool {
+		if bindingName != nil {
+			shootVertex, found := g.getVertex(VertexTypeShoot, shoot.Namespace, shoot.Name)
+			if !found {
+				return true
+			}
+			bindingVertex, found := g.getVertex(vertexType, shoot.Namespace, *bindingName)
+			if !found || !g.graph.HasEdgeFromTo(bindingVertex.id, shootVertex.id) {
+				return true
+			}
+		}
+		return false
```

Comment:
Nit: A bit more concise, WDYT?

```suggestion
		if bindingName == nil {
			return false
		}
		
		shootVertex, foundShootVertex := g.getVertex(VertexTypeShoot, shoot.Namespace, shoot.Name)
		bindingVertex, foundBindingVertex := g.getVertex(vertexType, shoot.Namespace, *bindingName)

		return !foundShootVertex || !foundBindingVertex || !g.graph.HasEdgeFromTo(bindingVertex.id, shootVertex.id)
```

Reply:
Nice improvement! Thanks!